### PR TITLE
[docs-beta] add Create a pipeline guide

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

Adds a Create a pipeline overview guide that walks through the 5 basic steps of building a pipeline. 

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by PedramNavid** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24307?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24307** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24305** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24305?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24304** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24304?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @PedramNavid and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/docs-beta/docs/guides/pipelines.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: "Create a pipeline"
+description: "Learn how to create data pipelines using Dagster's asset-based approach"
+---
+
+In Dagster, data pipelines are created using an asset-based approach. This overview will introduce you to the key concepts and steps involved in building a Dagster pipeline.
+
+## What you'll learn
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/pipelines.md:8]: Take this out, given that you're doing it elsewhere?

File: docs/docs-beta/docs/guides/pipelines.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+</details>
+
+Dagster uses software-defined assets as the building blocks of data pipelines. An asset represents a data object, such as a table, file, or machine learning model, that your pipeline produces or updates.
+
+As you define multiple assets and their dependencies, Dagster automatically creates an asset graph. This graph represents the structure and flow of your data pipeline. These nodes are not the individual operations that create the assets, but rather the assets themselves.
+
+```mermaid
+flowchart LR
+    A[Raw Sales Data] --> B[Cleaned Sales Data]
+    A -.-> C([Data Quality Checks])
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/pipelines.md:32]: Visually this struck me as a little odd. It took a second to realize that this wasn't actually an asset, even though it's a different shape and has a dashed connecting line vs solid.

- Should we differentiate this more?
- It's not mentioned in the steps below, which sorta makes it stand out to me - should it be included?

File: docs/docs-beta/docs/guides/pipelines.md (hunk 1 above)
Comment (PedramNavid) [reply docs/docs-beta/docs/guides/pipelines.md:8]: was removed in the upstream PR

**Review by erinkcochran87 — APPROVED** [review]

**Comment by PedramNavid** [comment]
### Merge activity

* **Sep 11, 12:23 PM EDT**: @PedramNavid started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24307).

**Pull request merged**